### PR TITLE
Fix service level negative timeouts

### PR DESCRIPTION
**Pull request by psarna**
This series fixes a minor validation issue with service level timeouts - negative values were not checked. This bug is benign because negative timeouts act just like a 0s timeout, but the original series claimed to validate against negative values, so it's hereby fixed.
More importantly however, this series follows by enabling cql-pytest to run service level tests and provides a first batch of them, including a missing test case for negative timeouts.
The idea is similar to what we already have in alternator test suite - authentication is unconditionally enabled, which doesn't affect any existing tests, but at the same time allows writing test cases which rely on authentication - e.g. service levels.

One very sad side effect of this change is +15 seconds of boot time, which is a result of a hardcoded delay in Scylla authentication module:
https://github.com/scylladb/scylla/blob/c0dafa75d9f334c1ac932aa7de5618ac5c80f188/auth/common.hh#L73-L75
While this delay is more or less negligible for production clusters, it's extremely annoing for cql-pytest, since 15 seconds is a heavy chunk of its total execution time. Perhaps it's time to try and reevaluate the hardcoded 15 seconds and try to fix the issue more properly? In particular, a local single node cluster should be able to set up its authentication tables almost immediately, and definitely doesn't need to unconditionally wait for 15 seconds before trying to do so.

### CODE REVIEW

**Comment by nyh** [comment]
> One very sad side effect of this change is +15 seconds of boot time, which is a result of a hardcoded delay in Scylla authentication module:
> 
> https://github.com/scylladb/scylla/blob/c0dafa75d9f334c1ac932aa7de5618ac5c80f188/auth/common.hh#L73-L75
> 
> While this delay is more or less negligible for production clusters, it's extremely annoing for cql-pytest, since 15 seconds is a heavy chunk of its total execution time. Perhaps it's time to try and reevaluate the hardcoded 15 seconds and try to fix the issue more properly? In particular, a local single node cluster should be able to set up its authentication tables almost immediately, and definitely doesn't need to unconditionally wait for 15 seconds before trying to do so.

Why is this a new problem for the service-levels test? Why didn't we experience this 15 second delay in cql-pytest and Alternator tests?

I think it's worse than annoying - it's unacceptable that running a single cql-pytest should take 15 seconds when it now takes no more than 2 seconds. I could barely accept that a specific single test will take 15 seconds because of some timeout in the code it tests - but not that running any single test will take 15 seconds because of the boot overhead - which until today took less than a second. At a minimum, if we think we need such a fixed timeout (I think it's a horrible idea), let's make it a parameter... `test/*/run currently` already uses `--ring-delay 0` to avoid a similarly useless fixed timeout in the gossiper, so maybe we need another option to avoid it in another place - if we can't get rid of it completely :-(

**Comment by psarna** [comment]
@nyh no no, the 15 second delay is a one-time even when **booting** Scylla, specifically because passing the `--authenticator PasswordAuthenticator` parameter triggers auth initialization, which has a hardcoded delay. It wasn't seen in:
 * alternator tests, because alternator auth is a separate concept from Scylla's password authenticator and thus doesn't trigger the delay
 * cql-pytest, because the password authenticator was disabled so far

And yes, providing a parameter similar to `ring-delay` as a workaround seems reasonable, especially for local tests. I can actually incorporate it into this series, since the 15 second delay is really annoying.

**Comment by nyh** [comment]
> @nyh no no, the 15 second delay is a one-time even when **booting** Scylla, specifically because passing the `--authenticator PasswordAuthenticator` parameter triggers auth initialization

Yes, I know, that's what I meant when I said running one test - any one test - will be delayed by 15 seconds. If I write a test for, say, some SELECT feature, I have to wait 15 seconds for it to run. It's a slippery slope. This is how we have what we have in dtest :-( Personally, I think this is unacceptable - not just annoying.

>     * cql-pytest, because the password authenticator was disabled so far

I see.

> > And yes, providing a parameter similar to `ring-delay` as a workaround seems reasonable, especially for local tests. I can actually incorporate it into this series, since the 15 second delay is really annoying.

Good. But maybe it's also a good time to think why this 15 second delay exists at all, and whether it should exist... I'm sure that it also delays each dtest by 15 seconds, it was just hard to notice in the typical 240 seconds it takes to run a do-nothing dtest :-(

**Comment by psarna** [comment]
By the way - the commit message from the change which introduced this unconditional delay mentioned that it follows Cassandra's implementation - so it's quite possible (but I haven't checked) that the cost is the same for booting a Cassandra cluster with authentication enabled.

**Comment by psarna** [comment]
> > @nyh no no, the 15 second delay is a one-time even when **booting** Scylla, specifically because passing the `--authenticator PasswordAuthenticator` parameter triggers auth initialization
> 
> Yes, I know, that's what I meant when I said running one test - any one test - will be delayed by 15 seconds. If I write a test for, say, some SELECT feature, I have to wait 15 seconds for it to run. It's a slippery slope. This is how we have what we have in dtest :-( Personally, I think this is unacceptable - not just annoying.

Right. For single tests I usually spawn my own cluster and run the test case against it, so the boot time is amortized. But I definitely see the point - we're operating on a local single-node cluster, so we shouldn't be punished by long delays meant to work around hypothetical issues on large distributed clusters.

**Comment by psarna** [comment]
I'll rebase on top of https://github.com/scylladb/scylla/pull/8649 once it's accepted, and the 15s problem would go away.

**Comment by psarna** [comment]
Rebased on top of master - the hardcoded 15s timeout is now gone, so this series is free of this burden and ready for review.

**Comment by psarna** [comment]
*ping*

File: test/cql-pytest/test_service_levels.py [hunk 1]
```diff
@@ -0,0 +1,70 @@
+# -*- coding: utf-8 -*-
+# Copyright 2020 ScyllaDB
```
Comment (nyh) [inline test/cql-pytest/test_service_levels.py:2]: 2021, but I agree 2020 is the year that never ended...

**Review by nyh — APPROVED** [review]

**Comment by nyh** [comment]
Thanks. Merged. I removed the paragraphs about the 15 second slowdown (which are no longer relevant) from the merge message, but left its mention in one of the patches so that I won't need to rebase that branch.
